lib/home_widget.dart

```diff
@@ -92,9 +101,160 @@ class HomeWidget {
+  //       });
+  //     }
+  //     return path;
+  //   } catch (e) {
+  //     throw Exception('Failed to save screenshot to app group container: $e');
+  //   }
+  // }
+
+  static Future renderFlutterWidget(
+    Widget widget, {
+    String fileName = 'screenshot',
+    String key = 'filename',
+    double? pixelRatio,
+    Size? logicalSize,
+  }) async {
+    /// finding the widget in the current context by the key.
+    final RenderRepaintBoundary repaintBoundary = RenderRepaintBoundary();
+
+    /// create a new pipeline owner
+    final PipelineOwner pipelineOwner = PipelineOwner();
+
+    /// create a new build owner
+    final BuildOwner buildOwner = BuildOwner(focusManager: FocusManager());
+
+    // Size logicalSize = ui.window.physicalSize / ui.window.devicePixelRatio;
+    // pixelRatio ??= ui.window.devicePixelRatio;
+    try {
+      final RenderView renderView = RenderView(
+        view: ui.PlatformDispatcher.instance.implicitView!,
+        child: RenderPositionedBox(
+            alignment: Alignment.center, child: repaintBoundary),
+        configuration: ViewConfiguration(
+          size: logicalSize ?? Size.zero,
+          devicePixelRatio: 1.0,
+        ),
+      );
+
+      /// setting the rootNode to the renderview of the widget
+      pipelineOwner.rootNode = renderView;
+
+      /// setting the renderView to prepareInitialFrame
+      renderView.prepareInitialFrame();
+
+      /// setting the rootElement with the widget that has to be captured
+      final RenderObjectToWidgetElement<RenderBox> rootElement =
+          RenderObjectToWidgetAdapter<RenderBox>(
+        container: repaintBoundary,
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Column(
+            // image is center aligned
+            mainAxisAlignment: MainAxisAlignment.center,
+            children: [
+              widget,
+            ],
+          ),
+        ),
+      ).attachToRenderTree(buildOwner);
+
+      ///adding the rootElement to the buildScope
+      buildOwner.buildScope(rootElement);
+
+      ///adding the rootElement to the buildScope
+      buildOwner.buildScope(rootElement);
+
+      /// finialize the buildOwner
+      buildOwner.finalizeTree();
+
+      ///Flush Layout
+      pipelineOwner.flushLayout();
+
+      /// Flush Compositing Bits
+      pipelineOwner.flushCompositingBits();
+
+      /// Flush paint
+      pipelineOwner.flushPaint();
+
+      final ui.Image image =
+          await repaintBoundary.toImage(pixelRatio: pixelRatio!);
+
+      /// The raw image is converted to byte data.
+      final ByteData? byteData =
+          await image.toByteData(format: ui.ImageByteFormat.png);
+
+      final PathProviderFoundation provider = PathProviderFoundation();
+
+      try {
+        final String? directory = await provider.getContainerPath(
+          appGroupIdentifier: HomeWidget.groupId!,
+        );
+        final String path = '$directory/$fileName.png';
```

Comment:
Would be nice to add a 'homeWidget' to the path.

```suggestion
        final String path = '$directory/homeWidget/$fileName.png';
        final File file = File(path);
        if (!(await file.exists())) {
          await file.create(recursive: true);
        }
```